editor/gui/editor_spin_slider.cpp

```diff
@@ -378,7 +379,7 @@ void EditorSpinSlider::_draw_spin_slider() {
 	TS->free_rid(num_rid);
 
 	if (!hide_slider) {
-		if (editing_integer) {
+		if (editing_integer && get_max() == get_min()) {
```

Comment:
I tested the changes and they work as expected on top of my PR. However, the fix for [#42809](https://github.com/godotengine/godot/issues/42809)
 affects all integer properties across the editor and project settings, and also removes the option to restore the up/down arrows when needed. It would be better to move this fix into a separate PR since it introduces a broader behavior change that likely needs its own discussion.